chainer/functions/connection/embed_id.py

```diff
@@ -81,16 +81,17 @@ def forward(self, inputs):
         else:
             if self.ignore_label is None:
                 cuda.elementwise(
-                    'T gy, int32 x, int32 n_out', 'raw T gW',
-                    'int w_ind[] = {x, i % n_out}; atomicAdd(&gW[w_ind], gy)',
+                    'T gy, S x, S n_out', 'raw T gW',
+                    'ptrdiff_t w_ind[] = {x, i % n_out};'
```

Comment:
How about `ptrdiff_t` -> `S`? It's guaranteed to be representable by `S`.

Reply:
`ptrdiff_t` is better for CuPy.

Reply:
OK, I agree with you because it's used as an index.